IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m

```diff
@@ -219,12 +219,12 @@ - (void)saveToPasteBoard:(NSURL *)url
 }
 
 + (BOOL)completeAcquireToken:(nullable NSURL *)resultURL
-           sourceApplication:(nonnull NSString *)sourceApplication
+           sourceApplication:(nullable NSString *)sourceApplication
        brokerResponseHandler:(nonnull MSIDBrokerResponseHandler *)responseHandler
 {
-    BOOL isBrokerResponse = [self isResponseFromBroker:sourceApplication];
-
-    if (!isBrokerResponse)
+    // sourceApplication could be nil, we want to return early if we know for sure response is not from broker
+    BOOL isNotFromBroker = sourceApplication && ![self isResponseFromBroker:sourceApplication];
```

Comment:
nit: let's do isFromBroker and do a !check on the if. just to get into habit of not creating something contradicting in first place.

Also, is this the right condition?  (i.e./ this means
isFromBroker = !sourceApplication || [self isResponseFromBroker] .
I am not sure if we can conclude isFromBroker is there is no sourceApplication.

Reply:
agree, addressed :)